spec/models/user_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+
+RSpec.describe User, type: :model do
+  it "メール、パスワードが有効な状態であること" do
+    user = build(:user)
+    expect(user).to be_valid
+    end
+  
+  it "メールアドレスがなければ無効であること" do
+    user = FactoryBot.build(:user, email:nil)
+    user.valid?
+    expect(user.errors[:email]).to include("入力されていません。")
+  end
+
+  it "重複したメールアドレスなら無効な状態であること" do
+  FactoryBot.create(:user, email:"test@example.com")
+  user = FactoryBot.build(:user ,email:"test@example.com")
+  user.valid?
+  expect(user.errors[:email]).to include("重複しています。")
+  end
+
+  it "パスワードが６文字であれば有効なこと" do
+  password6 = "123456"
+  user = build(:user, password: "123456", password_confirmation: "123456")
+  user.valid?
+  expect(user).to be_valid
+  end
```

Comment:
1. 文字数関係のテストだと `'a' * 6` みたいに書くことが多いです（パッと見て文字数が伝わりやすい）
  情報量は少ない方が読む側は理解しやすいので、なるべくシンプルにを心掛けると良きです。

2. password_confirmation はモデルではなく controller 側（device）で使用しているもので、factory_bot での生成時は controller を挟まないのでここでは不要です。

3. `user.valid?` と `be_valid` は同じことをしているので `user.valid?` は不要です。

まとめるとこの辺は↓のようになります。
```suggestion
  it "パスワードが６文字であれば有効なこと" do
    user = build(:user, password: 'a' * 6)
    expect(user).to be_valid
  end
```